wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt

```diff
@@ -116,8 +161,25 @@ class EditProfileActivity : BaseMenuActivity() {
             dashpayUserAvatar.setImageDrawable(defaultAvatar)
         }
 
-        aboutMe.setText(profile.publicMessage)
-        displayName.setText(profile.displayName)
+        about_me.setText(profile.publicMessage)
+        display_name.setText(profile.displayName)
+    }
+
+    // This method works well when it is called the first time,
+    // but when it is called after updating the profile, `it` is the old profile
+    private val dashPayProfileObserver = Observer<DashPayProfile?> {
+        if (it != null) {
+            dashPayProfile = it
+            showProfileInfo(it)
+        }
+    }
+
+    private fun setEditingState(isEditing: Boolean) {
+        if (!isEditing) {
+            editProfileViewModel.dashPayProfileData.observe(this, dashPayProfileObserver)
+        } else {
+            editProfileViewModel.dashPayProfileData.removeObserver(dashPayProfileObserver)
```

Comment:
The purpose of this method is to support the updating the profile if it was changed on another device.

However, one the user starts typing, then the observer is removed.  

Originally, I had the observer turned back on after the update operation was successful and finished, but then the observer is called with the previous values in the profile and then the screen would change all the text fields back to the previous values.  Though, if one goes back to the More screen, then he newly changed Display Name would show up.

For now, after editing, then the View is no longer live.  This may actually be a bad idea to have the view being live.  It is perhaps unlikely that a user would update their profile on two devices at the same time.